app-games/src/main/java/com/aptoide/android/aptoidegames/home/NewAppVersionPromotionalView.kt

```diff
@@ -0,0 +1,153 @@
+package com.aptoide.android.aptoidegames.home
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.clearAndSetSemantics
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import cm.aptoide.pt.download_view.presentation.DownloadUiState
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Downloading
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Error
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Install
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Installed
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Installing
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Migrate
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Outdated
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.ReadyToInstall
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Uninstalling
+import cm.aptoide.pt.download_view.presentation.DownloadUiState.Waiting
+import cm.aptoide.pt.download_view.presentation.rememberDownloadState
+import cm.aptoide.pt.feature_apps.data.App
+import cm.aptoide.pt.feature_apps.presentation.AppUiState
+import cm.aptoide.pt.feature_apps.presentation.rememberApp
+import cm.aptoide.pt.feature_home.domain.Bundle
+import com.aptoide.android.aptoidegames.R
+import com.aptoide.android.aptoidegames.appview.buildAppViewRoute
+import com.aptoide.android.aptoidegames.installer.presentation.AppIconWProgress
+import com.aptoide.android.aptoidegames.installer.presentation.InstallViewShort
+import com.aptoide.android.aptoidegames.installer.presentation.ProgressText
+import com.aptoide.android.aptoidegames.theme.AGTypography
+import com.aptoide.android.aptoidegames.theme.Palette
+
+@Composable
+fun NewAppVersionPromotionalView(bundle: Bundle, navigate: (String) -> Unit) {
+  bundle.view?.let {
+    val (uiState, _) = rememberApp(it)
+
+    when (uiState) {
+      is AppUiState.Idle -> NewAppVersionPromotional(uiState.app, bundle.title, navigate)
+      is AppUiState.Loading -> LoadingView()
+      is AppUiState.NoConnection,
+      is AppUiState.Error -> Unit
+    }
+  }
+}
+
+@Composable
+private fun NewAppVersionPromotional(app: App, label: String, navigate: (String) -> Unit) {
+  val downloadState = rememberDownloadState(app)
+  val isDownloading = isDownloading(downloadState)
```

Comment:
Let's wrap it in `remember` to avoid recalculation on recompositions with the same `downloadState`

Reply:
Mb even extract this as a `remember` function as it is already used as is in 2 places

Reply:
if i wrap it in remember it wont change because it didnt recompose